src/animations.jl

```diff
@@ -6,19 +6,56 @@ struct AnimationTrack{T}
     values::Vector{T}
 end
 
+function Base.insert!(track::AnimationTrack, frame::Integer, value)
+    i = searchsortedfirst(track.frames, frame)
+    insert!(track.frames, i, frame)
+    insert!(track.values, i, value)
+    return track
+end
+
+function Base.merge!(a::AnimationTrack, others::AnimationTrack...)
+    for other in others
+        @assert other.name == a.name
+        @assert other.jstype == a.jstype
+        for i in eachindex(other.frames, other.values)
+            # TODO: improve performance.
+            insert!(a, other.frames[i], other.values[i])
+        end
+    end
+    return a
+end
+
 @with_kw struct AnimationClip
     tracks::Dict{String, AnimationTrack} = Dict{String, AnimationTrack}()
     fps::Int = 30
     name::String = "default"
 end
 
+function Base.merge!(a::AnimationClip, others::AnimationClip...)
+    for other in others
+        @assert other.fps == a.fps
+        merge!(merge!, a.tracks, other.tracks) # merge tracks recursively
```

Comment:
I was just thinking: there could potentially be two animation tracks with the same name but different `jstype`s. That would currently error here. Again, not with the current `atframe` implementation,

https://github.com/rdeits/MeshCat.jl/blob/160933c0ae285384da15a9b653010f23046c137d/src/atframe.jl#L12-L14

but still.